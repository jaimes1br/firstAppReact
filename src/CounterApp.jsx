import { useState } from 'react'
import PropTypes from 'prop-types';


const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState(value);
  
  const handleAdd = (event) => {
    // console.log(event)
    setCounter(counter + 1);
  }

  const handleSubstract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);
  
  return (
    <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2> 
        {/* <button onClick={ (event) => handleAdd(event) }> //es lo mismo que la siguiente linea*/} 
        <button aria-label='btn-add' onClick={ handleAdd }>+1</button>
        <button aria-label='btn-sub' onClick={ handleSubstract }>-1</button>
        <button aria-label='btn-reset' onClick={ handleReset }>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp
