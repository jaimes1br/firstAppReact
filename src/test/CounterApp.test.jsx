import { fireEvent, render, screen} from "@testing-library/react";
import CounterApp from '../CounterApp';


describe('Test to <CounterApp/>', () => {

    const initialValue = 10;

    test('should match with snapshot', () => {
      
        const { container } = render( <CounterApp value={ initialValue }/> );
        expect( container ).toMatchSnapshot();
    });
    
  
    test('should show the initialValue 100 in the screen (<CounterApp value={100}/>)', () => {
        
        render(<CounterApp value={ 100 }/>);
        expect( screen.getByText(100)).toBeTruthy();
        expect (screen.getByRole('heading',{level: 2}).innerHTML).toBe(`${100}`);
    });

    test('should add +1 when press on buttom +1', () => {
       
        render(<CounterApp value={ initialValue }/>);
        // fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByRole('button',{ name: 'btn-add'}));
        expect(screen.getByText('11')).toBeTruthy();
    });
    
    
    test('should substract -1 when press on buttom -1', () => {
       
        render(<CounterApp value={ initialValue }/>);
        // fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByRole('button',{ name: 'btn-sub'}));
        expect(screen.getByText('9')).toBeTruthy();
    });

    
    test('should reset the value to initialValue', () => {
       
        render(<CounterApp value={ initialValue }/>);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByText('Reset'));
        fireEvent.click(screen.getByRole('button',{ name: 'btn-reset'}));
        expect(screen.getByText(initialValue)).toBeTruthy();
    });

});

