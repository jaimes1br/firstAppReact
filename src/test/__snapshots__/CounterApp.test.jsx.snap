// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <CounterApp/> should match with snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-sub"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
